fe/src/context/IssuesFilterContext.tsx

```diff
@@ -0,0 +1,179 @@
+import { IssuesFilter, IssuesFilterAction } from "@customTypes/index";
+import {
+  generateFilterText,
+  parseFilterTextToArray,
+} from "@utils/generateFilterText";
+import {
+  Dispatch,
+  ReactNode,
+  createContext,
+  useContext,
+  useReducer,
+} from "react";
+
+const IssuesFilterContext = createContext<{
+  issuesFilter: IssuesFilter;
+} | null>(null);
+
+const IssuesFilterDispatchContext =
+  createContext<Dispatch<IssuesFilterAction> | null>(null);
+
+export const INITIAL_FILTER_TEXT = "is:issue is:open";
+
+const initialFilter: IssuesFilter = {
+  state: {
+    status: "open",
+    filterBar: "open",
+    author: null,
+    assignees: new Set(),
+    labels: new Set(),
+    milestone: null,
+  },
+  text: INITIAL_FILTER_TEXT,
+};
+
+const issuesFilterReducer = (
+  issuesFilter: IssuesFilter,
+  action: IssuesFilterAction
+): IssuesFilter => {
+  switch (action.type) {
+    case "SET_AUTHOR": {
+      const newIssuesFilter = { ...issuesFilter };
+      newIssuesFilter.state.author = action.payload;
+      const updatedIssuesFilterText = generateFilterText(newIssuesFilter.state);
+
+      return { ...newIssuesFilter, text: updatedIssuesFilterText };
+    }
```

Comment:
return 부분에서 다시 복사하는 거면 newIssuesFilter 만들지 않고 요렇게도 가능할까요?
```suggestion
    case "SET_AUTHOR": {
      const updatedIssuesFilterText = generateFilterText(newIssuesFilter.state);

      return { ...issuesFilter, text: updatedIssuesFilterText, state: { author: action.payload } };
    }
```

Reply:
엇 그렇게 하게 되면 state가 author만 남게 될 것 같네요. 그리고 newIssuesFilter를 별도로 선언하지 않고 새로운 객체를 바로 넣게 되면 text를 만들기가 애매해져서, 선언하면서 바로 변경된 값도 넣어주도록 개선해보겠습니다!

```ts
case "SET_AUTHOR": {
      const newIssueFilterState = { ...issuesFilter.state, author: action.payload }
      const updatedIssuesFilterText = generateFilterText(newIssueFilterState);

      return {
        state: newIssueFilterState,
        text: updatedIssuesFilterText,
      };
    }
```